modules/api/common_types.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Service contains the fields that are needed for each OpenStack service CRD
+type Service struct {
+	// +kubebuilder:validation:Required
+	// The service specific Container Image URL
+	ContainerImage string `json:"containerImage"`
```

Comment:
Given external dataplane with mixed-rhel cases, shouldn't we make this a map, like today tripleo framework aims to handle OS-dependent containers images type? This could be pointless for native resources, but making sense for external Nova computes.

Reply:
This is part of the Spec of the Nova<ServiceType> CR to describe what container image should be running in the pod. What would it mean if there would be a map of images? Which container image should be running in the pod?

Reply:
as I said, that would have not much sens for native services, but only exnternal ones, managed with ansible operator.
Those will not be deployed as pods, but information in this CR for container image to put into ansible inventory could be important for openstack meta operator watching these CRs... if that makes sens

Reply:
well the top level NOVA CR need an optional map of container images for each nova service CR that is deployed beneath it 

at the controller side only one image is going to be supproted at a time since the mix host os support is only for external dataplne

for the compute nodes i expect thso to be modeled out side of the top level NovaCRD and i expect they will have an image specific per compute node service and that might also be a map

e.g. {nova-compute:rhel10-nova-compute, nova-libirvirt:rhel9-nova-libvirt}

Reply:
> well the top level NOVA CR need an optional map of container images for each nova service CR that is deployed beneath it

The Nova CR will have it via the service CR instances contained in them. There is no need for a separate map. 

> 
> at the controller side only one image is going to be supproted at a time since the mix host os support is only for external dataplne
> 
> for the compute nodes i expect thso to be modeled out side of the top level NovaCRD and i expect they will have an image specific per compute node service and that might also be a map
> 
> e.g. {nova-compute:rhel10-nova-compute, nova-libirvirt:rhel9-nova-libvirt}